networkx/algorithms/bipartite/tests/test_matrix.py

```diff
@@ -31,6 +31,10 @@ def test_biadjacency_matrix(self):
             assert M.shape[0] == tops[i]
             assert M.shape[1] == bots[i]
 
+    G = nx.empty_graph(2)
+    M = bipartite.biadjacency_matrix(G, [0])
+    assert M == sp.sparse.coo_matrix(([], ([], [])), shape=(1, 1), dtype=np.float64)
+
```

Comment:
Since this isn't in a test method, it's not actually being run as part of the test suite... if it were, it would fail as 1) biadjacency_matrix returns sparse arrays, not sparse matrices (on `main` at least) 2) the `==` operator does element-wise comparison and will raise sparse efficiency warnings for this case and 3) the output format of the sparse array would be csr, not coo, by default.

Reply:
Yeah not really sure what happened here. I moved the test into the suite and it passed, but it shouldn't have. I added a trivial test and tested it against the expected matrix. But ran into some trouble with comparing the sparse arrays. I ended up using numpy's array_equal with the dense versions of the arrays. But would love to know a better way if you've got one.